src/evict/evict_page.c

```diff
@@ -575,6 +575,18 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
     if (!modified)
         return (0);
 
+    /*
+     * If we trying to evict a dirty page that does not belong to history store(HS) and checkpoint
+     * is processing HS file, then avoid evicting the dirty non-HS page for now if the cache is
+     * already dominated by dirty HS content.
+     *
+     * Evicting a non-HS dirty page can generate even more HS content. As we can not evict HS pages
+     * while checkpoint is operating on the HS file, we can end up in a situation where we exceed
+     * the cache size limits.
+     */
+    if (modified && !WT_IS_HS(btree->dhandle) && __wt_cache_hs_dirty(session) &&
```

Comment:
How do you know the cache is dominated by hs content? This change I think effectively blocks evicting data store tables if checkpoint is working on the hs table. If it takes a long time to checkpoint the hs table, then we may run into cache pressure.

Reply:
`__wt_cache_hs_dirty(session)` returns true if HS content in cache > 20%

Reply:
OK, got it. Does 20% is high for hs?

Reply:
Note that `__wt_cache_hs_dirty()` is returning true when the dirty content from HS alone is 20% or more. We generally want to keep overall WT dirty content below 20% to keep the memory fragmentation to minimum.